File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-052112-30457/workdir/src/main/java/customerimporter/PersistentCustomerSystem.java
package customerimporter;

import java.util.Arrays;

public class PersistentCustomerSystem extends CustomerSystem {

    private DataBaseSession session;

    // customers

    public void add(Customer aCustomer) {
        session.persist(aCustomer);
    }

    public Customer customerWithIdentificationType(String anIdType, String anIdNumber) {
        return session
            .select(aCustomer -> aCustomer.isIdentifiedAs(anIdType, anIdNumber), Customer.class)
            .iterator().next();
    }

    public int numberOfCustomers() {
        return session.selectAllOfType(Customer.class).size();
    }

    // transactions

    public void beginTransaction() {
        session.beginTransaction();
    }

    public void commit() {
        session.commit();
    }

    // system lifecycle

    public void start() {
        session = DataBaseSession.forConfiguration(Arrays.asList(Address.class, Customer.class));
    }

    public void stop() {
        session.close();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/output/project/src/customerimporter/persistent_customer_system.py
from customerimporter.address import Address
from customerimporter.customer import Customer
from customerimporter.customer_system import CustomerSystem
from customerimporter.data_base_session import DataBaseSession


class PersistentCustomerSystem(CustomerSystem):

    _session = None

    # customers

    def add(self, a_customer):
        self._session.persist(a_customer)

    def customer_with_identification_type(self, an_id_type, an_id_number):
        return next(iter(self._session
            .select(lambda a_customer: a_customer.is_identified_by(an_id_type, an_id_number), Customer)))

    def number_of_customers(self):
        return len(self._session.select_all_of_type(Customer))

    # transactions

    def begin_transaction(self):
        self._session.begin_transaction()

    def commit(self):
        self._session.commit()

    # system lifecycle

    def start(self):
        self._session = DataBaseSession.for_configuration([Address, Customer])

    def stop(self):
        self._session.close()
